### ISSUE TITLE
Convert Existing MoleculeNet loaders to match template

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1999]
The new template in #1938 provides a powerful generalization of MoleculeNet. We should convert all existing loaders to this style.

CC @ncfrey @nd-02110114 @peastman

**Comment by ncfrey** [comment]
I am going to refactor the new materials datasets to match the template and then submit a PR for those loaders, so that will provide some examples of how to use it.

**Comment by nd-02110114** [comment]
@ncfrey 
When you try to add the new materials datasets to molnet, I'm really helpful if the dataset is Materials Project Dataset!

**Comment by ncfrey** [comment]
> @ncfrey
> When you try to add the new materials datasets to molnet, I'm really helpful if the dataset is Materials Project Dataset!

Yes, definitely! I have two datasets of over 100,000 structures with band gaps and formation energies from Materials Project that I plan to add.
